File: Arrays/Max_Consecutive_Ones/max_consecutive_ones.cpp
/**
 * Author: Nathan Bockisch
 * Date: September 29, 2021
 **/

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0, max_count = 0;

        for (int num : nums) {
            count += num;

            if (!num) {
                if (count > max_count)
                    max_count = count;
                count = 0;
                continue;
            }
        } 
        
        if (count > max_count)
            return count;
        
        return max_count;
    }
};

File: array_and_string/two_pointer_technique/max_consecutive_ones/max_consecutive_ones.rs
/*
 * Author: Nathan Bockisch
 * Date: July 12, 2022
 */
use std::cmp;

impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let (mut cur_count, mut max_count) = (0, 0);

        for num in nums {
            match num {
                1 => cur_count += 1,
                _ => {
                    max_count = cmp::max(cur_count, max_count);
                    cur_count = 0;
                }
            }
        }

        cmp::max(cur_count, max_count)
    }
}
